lib/room.cpp

```diff
@@ -557,6 +557,16 @@ QString Room::canonicalAlias() const
 
 QString Room::displayName() const { return d->displayname; }
 
+QList<const RoomEvent*> Room::pinnedEvents() const
+{
+    QStringList events = d->getCurrentState<RoomPinnedEvent>()->pinnedEvents();
+    QList<const RoomEvent*> pinnedEvents;
+    QStringList::iterator i;
+    for (i = events.begin(); i != events.end(); ++i)
+        pinnedEvents.append(findInTimeline(*i)->event());
```

Comment:
Before going with a proper review - what if `findInTimeline()` returns `historyEdge()` (in C++ terms, `deque::end()`) because the event isn't in the timeline yet?